### PR TITLE
Also removing bad tablets in FixShardReplication.

### PR DESCRIPTION
**Pull request by alainjobart**
Adding unit tests for FixShardReplication too.

### CODE REVIEW

File: go/vt/logutil/logger.go [hunk 1]
```diff
@@ -231,6 +231,13 @@ func (ml *MemoryLogger) String() string {
 	return buf.String()
 }
 
+// Clear clears the logs.
+func (ml *MemoryLogger) Clear() {
```
Comment (michael-berlin) [inline go/vt/logutil/logger.go]: optional: You could name this `ClearForTesting` to clarify that it should only be used by tests?

Comment (alainjobart) [inline go/vt/logutil/logger.go]: I don't think there is any reason a prod usage couldn't use it?

Comment (michael-berlin) [inline go/vt/logutil/logger.go]: Okay. Works for me. That's why I marked it as optional and LGTM'd the change overal :)

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**